### PR TITLE
[oboe] add new port

### PR DESCRIPTION
**Pull request by kafeg**

### CODE REVIEW

File: ports/oboe/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+else()
+    set(CMAKE_ARGS "-DBUILD_SHARED_LIBS=false")
+endif()
+
+vcpkg_cmake_configure(
+    SOURCE_PATH ${SOURCE_PATH}
+    OPTIONS
+        -DOBOE_EXAMPLES=OFF
+        ${CMAKE_ARGS}
+)
```
Comment (dg0yt) [inline ports/oboe/portfile.cmake:24]:
```suggestion
vcpkg_cmake_configure(
    SOURCE_PATH "${SOURCE_PATH}"
    OPTIONS
        -DOBOE_EXAMPLES=OFF
)
```

`BUILD_SHARED_LIBS` is set by vcpkg.
The stl type should be controlled by triplet/toolchain, not by portfile.
(And `CMAKE_ARGS` would need to be constructed as cmake list, not as string.)

File: ports/oboe/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+vcpkg_cmake_install()
+vcpkg_copy_pdbs()
+
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
```
Comment (dg0yt) [inline ports/oboe/portfile.cmake:29]:
```suggestion
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
```

File: ports/oboe/vcpkg.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "oboe",
+  "version": "1.7.0",
```
Comment (JonLiu1993) [inline ports/oboe/vcpkg.json:3]: @kafeg, Thanks for your PR, I noticed that the latest released version is 1.8.0, can you use the latest version?

File: ports/oboe/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,30 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO google/oboe
+    REF 1.7-stable
```
Comment (JonLiu1993) [inline ports/oboe/portfile.cmake:4]:
```suggestion
    REF "${VERSION}"
```

File: versions/o-/oboe.json [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  "versions": [
+    {
+      "git-tree": "c82ab85b41edab029b0b7153b62545539b7e31c4",
+      "version": "1.8.0",
+      "port-version": 0
+    },
+    {
+      "git-tree": "ee92a20e98ef519e9d67a522cb05619da27db1c1",
+      "version": "1.7.0",
+      "port-version": 0
```
Comment (JonLiu1993) [inline versions/o-/oboe.json:11]:
```suggestion

```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/oboe/portfile.cmake [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    SHA512 7eeaf85f9889e03dd1e7f5de0e9f2cee815fc555fddfdb8c4d3450d67f6ae11b0ca43b63c73e869bfc4629d2f8e5bdb23a5833c665ca5226c339f74b9b34a8ad
+    HEAD_REF master
+    PATCHES
+        fix_install.patch
+)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH ${SOURCE_PATH}
+    OPTIONS
+        -DOBOE_EXAMPLES=OFF
```
Comment (JonLiu1993) [inline ports/oboe/portfile.cmake:14]:
```suggestion
```

Comment (JonLiu1993) [reply ports/oboe/portfile.cmake:14]: `OBOE_EXAMPLES` does not seem to exist in the upstream, and it makes no sense to add it here.

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**